crates/re_log_types/Cargo.toml

```diff
@@ -76,7 +77,7 @@ ndarray.workspace = true
 nohash-hasher = "0.2"
 num-derive = "0.3"
 num-traits = "0.2"
-smallvec = "1.10"
+smallvec = { version = "1.10", features = ["union"] }
```

Comment:
we should add `smallvec` to the workspace `Cargo.toml`, including all opt-in features that are pure wins (and that only exist for backwards compatibility), including `union`